### DISCUSSION TITLE
Custom Policy with table join issue.

### DISCUSSION THREAD

**Discussion by bluebill1049** [supabase/supabase#2978] [Questions] [answered]
Hi there,

I have two tables

```
users
--------
id: uid
account_id: uid (foreign key to account's id)

accounts
--------
id: uid
```

My policy blow, for some reason it is not working... however, if I run the SQL in the query tool and it's fine by hard code the `uid`.

❌ Fails with policy:  (even if i hard code the uid)
```sql
EXISTS (
  SELECT
    t.id,
    t.account_id,
    a.name
  FROM
    (
      users t
      INNER JOIN accounts a ON ((a.id = t.account_id))
    )
  WHERE
    (
      t.id = uid();
    )
)
```

✅ Works: 
```sql 
EXISTS (
  SELECT
    t.id,
    t.account_id
  FROM
    users t
  WHERE
      t.id = uid();
)
```

The strange behavior is when I remove the table join and then the policy works. As if it's only working without the join, hope to get some insight into why is this happening and potentially the root cause. Quite new to policy at the moment.

reference from this question: https://github.com/supabase/supabase/discussions/811

Thanks in advance.

**Comment by w3b6x9** [comment, marked as answer]
@bluebill1049 do you have any policies setup on the `accounts` table? This seems to be a permissions issue so you might need to update any existing policies on `accounts` or create a security definer function and use that as part of your policy. See [https://supabase.io/docs/guides/auth#policies-with-security-definer-functions](https://supabase.io/docs/guides/auth#policies-with-security-definer-functions) for more information.

**Reply by bluebill1049** [reply]
oh, that's a very good point! This makes sense 👍 I will test out and let you know.

**Reply by bluebill1049** [reply]
Thank you very much @w3b6x9 🙏 That's the root cause. you are a lifesaver.
